### PR TITLE
STOPGAP MEASURE - panic when too many PeriodicAppClientTasks are hanging

### PR DESCRIPTION
**Pull request by gvaradarajan**
There is currently an instability on ESP32s that we believe is being caused by a request to the Viam App properly resolving but hanging on dropping the TLS context. This causes signaling to hang and blocks all incoming requests rendering the machine inaccessible and unable to move forward with any of its other tasks. The purpose of this PR is two-fold: we want to collect more data around the frequency with which these tasks hang and when all the tasks are in such a state we want to prompt a restart that will start of all the tasks anew. 

This is intended to be temporary until the investigation resolves and a proper fix is scoped out.

### CODE REVIEW

File: micro-rdk/src/common/conn/server.rs [hunk 1]
```diff
@@ -470,6 +474,16 @@ where
...
                         .await
                         .insert(app_client);
+                } else {
+                    let current_count = urguard.as_ref().unwrap().get_grpc_client_count();
+                    log::info!(
+                        "ViamServer signaling flow has created grpc client clone {} of {}",
+                        current_count,
+                        max_anticipated_grpc_clients
+                    );
+                    if current_count > (max_anticipated_grpc_clients - 1) {
```
Comment (mattjperez) [inline micro-rdk/src/common/conn/server.rs:484]: nit: might be worth just repeating the `self.app_client_tasks.len()` here. don't have to think about what the -1 when looking at this line

**Review by mattjperez — APPROVED** [review]
LGTM

File: micro-rdk/src/common/conn/server.rs (hunk 1 above)
Comment (acmorrow) [reply micro-rdk/src/common/conn/server.rs:484]: Or just write it as `= max...`. Or `>=` if you want to be a little paranoid.

**Review by acmorrow — APPROVED** [review]
LGTM

**Pull request merged**